org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/util/JavaDocHelper.xtend

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xbase.ui.util
+
+import com.google.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.jface.text.IRegion
+import org.eclipse.jface.text.Region
+import org.eclipse.xtext.common.types.TypesPackage
+import org.eclipse.xtext.documentation.IJavaDocTypeReferenceProvider
+import org.eclipse.xtext.naming.IQualifiedNameConverter
+import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.scoping.IScopeProvider
+import org.eclipse.xtext.util.Pair
+import org.eclipse.xtext.util.Tuples
+
+import static extension org.eclipse.xtext.nodemodel.util.NodeModelUtils.findActualSemanticObjectFor
+import static extension org.eclipse.xtext.nodemodel.util.NodeModelUtils.findLeafNodeAtOffset
+
+/**
+ * @author miklossy - Initial contribution and API
+ * 
+ * @since 2.16
+ */
+class JavaDocHelper {
+
+	@Inject extension IScopeProvider
+
+	@Inject extension IJavaDocTypeReferenceProvider
+
+	@Inject extension IQualifiedNameConverter
+
+	/**
+	 * Returns the referenced elements in javadoc comment at a given <i>offset</i>.
+	 */
+	def Pair<EObject, IRegion> getReferencedElementInJavaDoc(XtextResource resource, int offset) {
```

Comment:
If we introduce a new abstraction here with the JavaDocHelper, I'd like it to return a "better" type than a pair. Even if it is only a small data object that has an EObject and a region, it would make sense to encapsulate and name that better than `Pair<EObject, IRegion>`.